kratos/spatial_containers/tree.h

```diff
@@ -182,93 +176,116 @@ class Tree
+    /// The iterator type definition
+    using IteratorType = typename PartitionType::IteratorType;
 
-    typedef typename PartitionType::IteratorType IteratorType;
+    /// The distance iterator type definition
+    using DistanceIteratorType = typename PartitionType::DistanceIteratorType;
 
-    typedef typename PartitionType::DistanceIteratorType DistanceIteratorType;
+    /// The pointer type definition
+    using PointerType = typename PartitionType::PointerType;
 
-    typedef typename PartitionType::PointerType PointerType;
+    /// The distance function type definition
+    using DistanceFunction = typename PartitionType::DistanceFunction;
 
-    typedef typename PartitionType::DistanceFunction DistanceFunction;
+    /// Dimension definition
+    static constexpr std::size_t Dimension = PartitionType::Dimension;
 
-    enum { Dimension = PartitionType::Dimension };
+    /// The node type definition
+    using NodeType = TreeNode<Dimension,PointType,PointerType,IteratorType,DistanceIteratorType> ;
 
-    typedef TreeNode<Dimension,PointType,PointerType,IteratorType,DistanceIteratorType> NodeType;
+    /// The coordinate type definition
+    using CoordinateType = typename NodeType::CoordinateType;
 
-    typedef typename NodeType::CoordinateType CoordinateType;
+    /// The size type definition
+    using SizeType = typename NodeType::SizeType;
 
-    typedef typename NodeType::SizeType       SizeType;
+    /// The index type definition
+    using IndexType = typename NodeType::IndexType;
 
-    typedef typename NodeType::IndexType      IndexType;
-
-    //typedef typename NodeType::SearchStructureType SearchStructureType;
-    typedef typename PartitionType::SearchStructureType SearchStructureType;
+    /// The search structure type definition
+    using SearchStructureType = typename PartitionType::SearchStructureType;
 
     ///@}
     ///@name Life Cycle
     ///@{
 
-    /// Constructor.
-    Tree(IteratorType PointsBegin, IteratorType PointsEnd, SizeType BucketSize = 1)
-        : mBucketSize(BucketSize), mPointsBegin(PointsBegin), mPointsEnd(PointsEnd)
+    /**
+     * @brief Construct a new Tree object
+     * @param PointsBegin Iterator to the first point
+     * @param PointsEnd Iterator to the last point
+     * @param BucketSize Size of the bucket
+     */
+    Tree(
+        IteratorType PointsBegin,
+        IteratorType PointsEnd,
+        SizeType BucketSize = 1
+        ) : mBucketSize(BucketSize),
+            mPointsBegin(PointsBegin),
+            mPointsEnd(PointsEnd)
     {
-
         if(mPointsBegin == mPointsEnd)
             return;
-        PointType max_point;
-        PointType min_point;
-        for(SizeType i = 0 ; i < Dimension ; i++)
-        {
-            max_point[i] = (**mPointsBegin)[i];
-            min_point[i] = (**mPointsBegin)[i];
+
+        for(SizeType i = 0 ; i < Dimension ; i++) {
+            mBoundingBoxHighPoint[i] = (**mPointsBegin)[i];
```

Comment:
If I remove them I should remove the member variables and access methods, certainly I prefer this way, because it gives you additional info of the search

Reply:
And the fact those variables are there give me the impression that this was the original intent.

Reply:
Well, I would not keep them if they are not used but I get your point. Ok to me.